calblueprint/siren
# Phoebe/context provider

# ✨ New in this PR ✨

## One liner: what did you do? 🚨 
User Context Provider to persist logged-in client throughout app ([documentation](https://www.notion.so/calblueprint/Context-Provider-9406e7564ce24a9bb866a7a5af0d9f5f))

## Coverage 🙆‍♀️
_use this section to break up your task into submodules and track progress_ 
 - [x] create src/context/ContextProvider.tsx
 - [x] wrap app with ClientProvider in App.tsx
 - [x] update context in RootNavigator (since it handles user auth)
 - [x] remove/refactor uses of getCurrentClient()

## How can the reviewer test your code? 👩‍💻
- app should work as normal
- use in any React component (see example): 
    -  state is a Client object and update is a function that updates state
```  
import { ClientContext } from 'context/ContextProvider';
...
const { state, update } = React.useContext(ClientContext);
````

## Any bugs you encountered or still having trouble with? 🐛


## Resources 📔
https://react-typescript-cheatsheet.netlify.app/docs/basic/getting-started/context/
- I mainly copied the structure of extended example list item 4

🧜‍♀️ cc: @shannonbonet @phoebeli23
